--- a/src/opts/bitops.isle
+++ b/src/opts/bitops.isle
@@ -157,32 +157,3 @@
         (iconst_u ty 0xff00))
       (ushr ty x (iconst_u ty 24)))))
   (bswap ty x))
-
-(rule (simplify (bor ty @ $I64
-    (bor ty
-      (bor ty
-        (ishl ty x (iconst_u ty 56))
-        (ishl ty
-          (band ty x (iconst_u ty 0xff00))
-          (iconst_u ty 40)))
-      (bor ty
-        (ishl ty
-          (band ty x (iconst_u ty 0xff_0000))
-          (iconst_u ty 24))
-        (ishl ty
-          (band ty x (iconst_u ty 0xff00_0000))
-          (iconst_u ty 8))))
-    (bor ty
-      (bor ty
-        (band ty
-          (ushr ty x (iconst_u ty 8))
-          (iconst_u ty 0xff00_0000))
-        (band ty
-          (ushr ty x (iconst_u ty 24))
-          (iconst_u ty 0xff_0000)))
-      (bor ty
-        (band ty
-          (ushr ty x (iconst_u ty 40))
-          (iconst_u ty 0xff00))
-        (ushr ty x (iconst_u ty 56))))))
-  (bswap ty x))
