--- a/lib.rs
+++ b/lib.rs
@@ -55,7 +55,7 @@ use zerocopy::{AsBytes, FromBytes, FromZeroes};
 
 pub mod apple_codesign;
 
-#[cfg(target_os = "linux")]
+#[cfg(any(target_os = "linux", target_os = "android"))]
 pub use elf::find_section;
 #[cfg(target_os = "macos")]
 pub use macho::find_section;
@@ -737,7 +737,7 @@ impl<'a> Elf<'a> {
     }
 }
 
-#[cfg(target_os = "linux")]
+#[cfg(any(target_os = "linux", target_os = "android"))]
 mod elf {
     use std::io::Read;
     use std::io::Seek;
