--- a/v8/src/handles/handles.h
+++ b/v8/src/handles/handles.h
@@ -123,22 +123,11 @@ class Handle final : public HandleBase {
   // This means that this is only permitted for Tagged<T> with an operator->,
   // i.e. for on-heap object T.
   V8_INLINE Tagged<T> operator->() const {
-    if constexpr (is_subtype_v<T, HeapObject>) {
-      return **this;
-    } else {
-      // `static_assert(false)` in this else clause was an unconditional error
-      // before CWG2518. See https://reviews.llvm.org/D144285
-#if defined(__clang__) && __clang_major__ >= 17
-      // For non-HeapObjects, there's no on-heap object to dereference, so
-      // disallow using operator->.
-      //
-      // If you got an error here and want to access the Tagged<T>, use
-      // operator* -- e.g. for `Tagged<Smi>::value()`, use `(*handle).value()`.
-      static_assert(
-          false,
-          "This handle does not reference a heap object. Use `(*handle).foo`.");
-#endif
-    }
+    static_assert(
+        is_subtype_v<T, HeapObject>,
+        "This handle does not reference a heap object. Use `(*handle).foo`.");
+
+    return **this;
   }
 
   V8_INLINE Tagged<T> operator*() const {
@@ -409,19 +398,11 @@ class DirectHandle final : public DirectHandleBase {
                                                   : kTaggedNullAddress) {}
 
   V8_INLINE Tagged<T> operator->() const {
-    if constexpr (std::is_base_of_v<HeapObject, T> ||
-                  std::is_convertible_v<T*, HeapObject*>) {
-      return **this;
-    } else {
-      // For non-HeapObjects, there's no on-heap object to dereference, so
-      // disallow using operator->.
-      //
-      // If you got an error here and want to access the Tagged<T>, use
-      // operator* -- e.g. for `Tagged<Smi>::value()`, use `(*handle).value()`.
-      static_assert(
-          false,
-          "This handle does not reference a heap object. Use `(*handle).foo`.");
-    }
+    static_assert(
+      std::is_base_of_v<HeapObject, T> || std::is_convertible_v<T*, HeapObject*>,
+      "This handle does not reference a heap object. Use `(*handle).foo`.");
+
+    return **this;
   }
 
   V8_INLINE Tagged<T> operator*() const {
