--- a/build/config/android/BUILD.gn
+++ b/build/config/android/BUILD.gn
@@ -6,7 +6,6 @@
 import("//build/config/android/rules.gni")
 import("//build/config/c++/c++.gni")
 import("//build/config/compiler/compiler.gni")
-import("//build/config/gclient_args.gni")
 import("//build/config/sanitizers/sanitizers.gni")
 
 if (current_toolchain == default_toolchain) {
--- a/build/config/android/config.gni
+++ b/build/config/android/config.gni
@@ -158,6 +158,7 @@
   }
 
   android_ndk_root = "//third_party/android_toolchain/ndk"
+  android_ndk_version = "r28"
 
   public_android_sdk_root = "//third_party/android_sdk/public"
   public_android_sdk_platform_version = "37.0"
