--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -628,7 +628,7 @@
     # The performance improvement does not seem worth the risk. See
     # https://crbug.com/484082200 for background and https://crrev.com/c/7593035
     # for discussion.
-    if (!is_wasm) {
+    if (!is_wasm && !is_clang) {
       cflags += [ "-fno-lifetime-dse" ]
     }
 
@@ -1897,7 +1897,7 @@
       # sanitizer) is enabled, they then do expensive debug like operations. We
       # want to suppress this behaviour since we want to keep performance costs
       # as low as possible while having these checks.
-      "-fsanitize-ignore-for-ubsan-feature=array-bounds",
+      # "-fsanitize-ignore-for-ubsan-feature=array-bounds",
 
       # Because we've enabled array-bounds sanitizing we also want to suppress
       # the related warning about "unsafe-buffer-usage-in-static-sized-array",
