--- a/build/config/clang/BUILD.gn
+++ b/build/config/clang/BUILD.gn
@@ -268,32 +268,12 @@
 
 # Adds a dependency on the Clang runtime library clang_rt.builtins.
 clang_lib("compiler_builtins") {
-  if (is_mac) {
-    libname = "osx"
-  } else if (is_ios) {
-    if (target_platform == "iphoneos") {
-      if (target_environment == "simulator") {
-        libname = "iossim"
-      } else if (target_environment == "device") {
-        libname = "ios"
-      } else if (target_environment == "catalyst") {
-        libname = "osx"
-      } else {
-        assert(false, "unsupported target_environment=$target_environment")
-      }
-    } else if (target_platform == "tvos") {
-      if (target_environment == "simulator") {
-        libname = "tvossim"
-      } else if (target_environment == "device") {
-        libname = "tvos"
-      } else {
-        assert(false, "unsupported target_environment=$target_environment")
-      }
-    } else {
-      assert(false, "unsupported target_platform=$target_platform")
-    }
-  } else if (current_cpu != "ppc64" && current_cpu != "s390x") {
-    libname = "builtins"
+  if (is_android) {
+    libs = [ "$clang_base_path/lib/clang/$clang_version/lib/linux/libclang_rt.builtins-aarch64-android.a" ]
+  } else if (is_linux) {
+    libs = [ "/usr/lib/llvm-$clang_version/lib/clang/$clang_version/lib/linux/libclang_rt.builtins-x86_64.a" ]
+  } else {
+    assert(false)
   }
 }
 
